### PR TITLE
Fix jittering sounds on entities (fixes #2974)

### LINKED ISSUE #2974: Very fast repeating step sounds on entity collision
Standing on an entity forces it to "jiggle" and producing step sounds.
This can be reproduced by this three simple steps:

1.) place a boat on water (does also work with other entities)
2.) replace the water source with another node (for example: dirt)
3.) now step on the boat and listen :smile: 

Because I'm very bad at explaining things I made a short video:
https://youtu.be/EAwQ4Mwr4tc

### PR DESCRIPTION
**Pull request by BlockMen**
As title says and some codestyle fixes.

### CODE REVIEW

**Comment by PilzAdam** [comment]
Is this a proper fix? This doesn't seem to fix the jittering of `touching_ground`, which also causes other problems (e.g. jumping on top of entities randomly doesn't work due to this).

Also don't mix code style fixes with other commits.

**Comment by BlockMen** [comment]
@PilzAdam Yes and no. Yes, since it should never play a sound. And even if it should, then not the sound of the node the entity is located.
No, since there seems to be something wrong in general with object collision, but that the PR not claiming to fix ;)

Do we have a rule for that? But even if not, I will seperate it when someone else claims it has to be done.

File: src/collision.cpp [hunk 1]
```diff
@@ -537,23 +519,20 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
...
-		if(
-				cbox.MaxEdge.X-d > box.MinEdge.X &&
-				cbox.MinEdge.X+d < box.MaxEdge.X &&
-				cbox.MaxEdge.Z-d > box.MinEdge.Z &&
-				cbox.MinEdge.Z+d < box.MaxEdge.Z
-		){
-			if(is_step_up[boxindex])
-			{
+		if (cbox.MaxEdge.X - d > box.MinEdge.X && cbox.MinEdge.X + d < box.MaxEdge.X &&
+				cbox.MaxEdge.Z - d > box.MinEdge.Z &&cbox.MinEdge.Z + d < box.MaxEdge.Z) {
```
Comment (est31) [inline src/collision.cpp]: This line is too long, it has 91 chars.

File: src/collision.cpp [hunk 2]
```diff
@@ -509,10 +492,10 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 			}
 
 			info.new_speed = speed_f;
-			if(info.new_speed.getDistanceFrom(info.old_speed) < 0.1*BS)
+			if (info.new_speed.getDistanceFrom(info.old_speed) < 0.1 * BS)
 				is_collision = false;
 
-			if(is_collision){
+			if (is_collision){
```
Comment (rubenwardy) [inline src/collision.cpp]: missed brace here

File: src/collision.cpp (hunk 1 above)
Comment (est31) [inline src/collision.cpp]: And there is a space needed after the &&.

**Comment by est31** [comment]
Looks good.

**Pull request merged**